src/test/java/http_server/ServerSpy.java

```diff
@@ -0,0 +1,33 @@
+package http_server;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ServerSpy extends Server {
+
+    private RequestParser requestParser = new RequestParser();
+    private ResponseMaker responseMaker = new ResponseMaker();
+    private List<Boolean> possibleOptions = new LinkedList<>(Arrays.asList(true, false));
+
+    public ServerSpy(String host, int port) {
```

Comment:
Is this truly a spy?  There's a lot going on in this class, it's very complicated.  So much, in fact, that you might want tests for this.  Can it be simplified?